docs/scripts/formattedTSDemos.js

```diff
@@ -106,8 +106,15 @@ async function transpileFile(tsxPath, program, ignoreCache = false) {
     }
     const { code } = await babel.transformAsync(source, transformOptions);
 
+    const prettierConfigPath = await prettier.resolveConfigFile();
+    if (!prettierConfigPath) {
+      throw new Error(
+        `Could not resolve prettier config file.
+        Please provide a valid prettier config path or ensure that a prettier config file exists in the project root.`,
+      );
+    }
```

Comment:
```suggestion
    const prettierConfigPath = await resolvePrettierConfigPath()
```

Reply:
Tried it: https://github.com/mui/mui-x/pull/18720/commits/03d59e90578b8f307fab1f4c871bf03084812ca4
I wanted to limit the scope of changes.
For that to work we need to unify file extensions or move to tsx for execution, but it seems that the latest "trend" is towards going for `mjs`, so I didn't want to do it. 🤔